### DISCUSSION TITLE
Specify groupBy column order in plotPeak2GeneHeatmap()

### DISCUSSION THREAD

**Discussion by pseudacriscrucifer** [GreenleafLab/ArchR#2077] [Questions / Documentation]
Is there a way to sort the groupBy groups for a peak-to-gene heatmap such that columns reoresenting the groupBy group are grouped together completely, and not broken up across the resulting heatmap?

For example, in the attached image of peak-to-gene association heatmap, the groupBy row indicating cell type identity, how might I use plotPeak2GeneHeatmap() to organize neatly this row (i.e. more like the left portion of the groupBy row with consistent cell type identity, opposed to the right side of this row where that is much more broken up).

The only parameter I have changed has been groupBy, but I cannot find anything else to change to alter the order of the groupBy groups.

![groupByQ](https://github.com/GreenleafLab/ArchR/assets/93164822/4898ffe4-7b72-4d32-956d-6479606ddd7d)

**Comment by liangliangjin** [comment]
Have you finally solved this problem? I also want to know how to achieve this, especially if I have a multi-tissue proj that includes a very large number of cell types. I know that many heat map packages can cluster row and column labels at the same time, but archr data is not a simple matrix, making it difficult to modify your code.
